src/main/java/com/stripe/net/ApiResource.java

```diff
@@ -156,6 +156,20 @@ public static String urlEncode(String str) throws UnsupportedEncodingException {
     }
   }
 
+  /**
+   * URL-encode a string ID in url path formatting.
+   */
+  public static String urlEncodeId(String id) throws InvalidRequestException {
+    try {
+      return urlEncode(id);
+    } catch (UnsupportedEncodingException e) {
+      throw new InvalidRequestException(String.format(
+          "Unable to encode `%s` in the url to %s. "
+          + "Please contact support@stripe.com for assistance.", id, CHARSET),
+          null, null, null, 0, e);
+    }
```

Comment:
Not wrapping this checked `UnsupportedEncodingException` and breaks existing integration.
Existing integration however already has checked `InvalidRequestException` in its method signature.